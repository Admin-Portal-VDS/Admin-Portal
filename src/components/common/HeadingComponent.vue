<template>
  <div class="heading">{{ heading }}</div>
</template>

<script>
export default {
  name: 'HeadingComponent',
  props: {
    heading: {
      type: String,
      required: true,
    },
  },
};
</script>

<style scoped>
.heading {
  font-size: 30px;
  margin-bottom: 40px;
}
</style>
