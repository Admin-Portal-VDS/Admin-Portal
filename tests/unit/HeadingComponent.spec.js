import { mount } from '@vue/test-utils';
import HeadingComponent from '@/components/common/HeadingComponent.vue';

describe('HeadingComponent.vue', () => {
  it('renders the heading with the correct text', () => {
    const heading = 'Hello World';
    const wrapper = mount(HeadingComponent, {
      props: {
        heading,
      },
    });
    const title = wrapper.find('.heading');
    expect(title.exists()).toBe(true);
    expect(title.text()).toBe(heading);
  });
  it('accepts props and passes them correctly', () => {
    const heading = 'Test Button';
    const wrapper = mount(HeadingComponent, {
      propsData: { heading },
    });
    expect(wrapper.props('heading')).toBe(heading);
  });

  it('throws a warning if required prop is missing', () => {
    const warnSpy = jest.spyOn(console, 'warn').mockImplementation(() => {});
    mount(HeadingComponent, {});
    expect(warnSpy).toHaveBeenCalledTimes(1);
    expect(warnSpy.mock.calls[0][0]).toEqual(
      expect.stringContaining('Missing required prop: "heading"'),
    );
  });

  it('matches snapshot', () => {
    const wrapper = mount(HeadingComponent, {
      propsData: {
        heading: 'Snapshot Test', // Correct prop name
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
