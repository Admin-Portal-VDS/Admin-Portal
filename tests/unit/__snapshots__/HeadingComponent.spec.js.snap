// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingComponent.vue matches snapshot 1`] = `<div class="heading">Snapshot Test</div>`;
